vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -405,6 +409,80 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
             }
           }
 
+          // http2 pushing support
+          if (request.version() == HttpVersion.HTTP_2 && http2PushMapping != null) {
+
+            JsonArray dependencies = http2PushMapping.getJsonArray(file.substring(webRoot.length() + 1));
+            if (dependencies != null) {
+
+              for(Object dependency : dependencies) {
```

Comment:
This can be dangerous since we loop all dependencies and push, if the number is high we might strangle the connection, we need to have a smart way to do this? batch? limit?

Reply:
This is indeed true, could you explain a bit more about how I can do it?